internal/bgp/native/native.go

```diff
@@ -0,0 +1,708 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package native
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"hash/crc32"
+	"io"
+	"io/ioutil"
+	"net"
+	"os"
+	"sync"
+	"syscall"
+	"time"
+	"unsafe"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"go.universe.tf/metallb/internal/bgp"
+	"golang.org/x/sys/unix"
+)
+
+var errClosed = errors.New("session closed")
+
+// Session represents one BGP session to an external router.
+type Session struct {
```

Comment:
This can be private now I think

Reply:
Actually both `session` and `sessionManager` types can be private in both `native.go` and `frr.go`.